tests-scan

```diff
@@ -205,14 +205,47 @@ def queue_test(job: Job, channel, options: argparse.Namespace) -> None:
     if command:
         priority = min(job.priority, distributed_queue.MAX_PRIORITY)
 
+        slug = f"{job.name}-{job.revision[:8]}-{time.strftime('%Y%m%d-%H%M%S')}-{job.context.replace('/', '-')}"
+        (image, _, scenario) = job.context.partition("/")
+
+        env = {
+            "TEST_OS": image,
+        }
+        if scenario:
+            env["TEST_SCENARIO"] = scenario
+        if job.bots_ref:
+            env["COCKPIT_BOTS_REF"] = job.bots_ref
+        if job.base:
+            env["BASE_BRANCH"] = job.base
+
         body = {
+            # old-style command line jobs
             "command": command,
             "type": "test",
             "sha": job.revision,
             "ref": job.ref,
             "name": job.name,
-            "container": job.container,
+            # future job-runner format; once everything moved over, remove the stuff above
+            "job": {
+                "repo": job.repo,
+                "sha": job.revision,
+                "context": job.context,
+                "target": job.base,
+                "slug": slug,
+                "env": env,
+                "container": job.container,
+                # for updating naughty trackers
+                "secrets": ["github-token"],
+            }
         }
+
+        # report issue for nightly runs
+        if job.number == 0:
+            body["job"]["report"] = {
+                "title": f"Tests failed on {job.revision}",
+                "labels": ["nightly"],
+            }
```

Comment:
In case `job.number` is set to a valid PR number, can you set `body['job']['pull']` accordingly?  I'm going to add a feature to `job-runner` to support:
 1. specifying the test subject via PR number instead of (or in addition to) sha; and
 2. if both are specified, polling the specified PR (instead of the problematic commit/x/pulls API) to verify that the head branch is still equal to the provided SHA

Reply:
Right, makes sense. Added, covered by unit tests, and I also checked it manually:
```
./tests-scan -n -p 5932 --amqp foo:123
```

gives a job:
```json
{
    "repo": "cockpit-project/bots",
    "sha": "4c2c7f2c1c622ab0d46b91d24c8205a47aff2fd8",
    "context": "fedora-39",
    "target": "main",
    "slug": "pull-5932-4c2c7f2c-20240228-110700-fedora-39",
    "env": {
        "TEST_OS": "fedora-39",
        "BASE_BRANCH": "main"
    },
    "container": null,
    "secrets": [
        "github-token"
    ],
    "pull": 5932
}
```